### PR TITLE
Allow frontend filtering by neighborhood

### PR DESCRIPTION
**Pull request by abbiefarr**
## Issue
#55 

Addresses # (issue)
- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Implements filtering by property name through the filters modal. Other filters are not implemented yet.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?
Click the "Filter Listings" button in the public portal. Try filtering by "Property Name"

Please describe the tests that you ran to verify your changes. Provide instructions so we can review. Please also list any relevant details for your test configuration
- [x] Desktop View
- [x] Mobile View

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes

### CODE REVIEW

**Review by avaleske — CHANGES_REQUESTED** [review]
Thanks for getting this connected and working!

I noted in the comments, but I think we should use neighborhood, not name, even for this initial version. We can add a ticket for adding an endpoint to get neighborhoods from the db, but for now a hardcoded list is fine I think.

Maybe out of scope for this, but when it comes time to try to update the URL with the applied filter, I think we have a few options. I drew some diagrams to help me think about what was happening, and I figured they'd be helpful to share since they apply for the page state too:
- `onSubmit()` just updates the URL, which then gets decoded to update the filter/page state (Image 1)
- `onSubmit()` updates both the URL and the filter/page state (Image 2)

Right now, this PR just doing the update in blue, not the top loop, for filters. The update logic is doing both loop for pages, which is why we have to check if `page != currentPage`.

Image 1 (Update just the URL)


Image 2 (Update both the URL and the page state)


Implicit in this is that the data call uses the same url format as the /listings url, so that encoding/decoding can re-use the same methods.

File: ui-components/src/locales/general.json [hunk 1]
```diff
@@ -1225,6 +1225,13 @@
         "senior": "Senior",
         "assisted": "Assisted"
       }
+    },
+    "nameOptions":{
+      "label": "Property Name:",
+      "nameOptionsTypes": {
+        "triton": "Test: Triton",
+        "coliseum": "Test: Coliseum"
+      }
```
[the remark this answers is not in the document]
Comment (abbiefarr) [reply ui-components/src/locales/general.json]: Done

File: sites/public/lib/hooks.ts [hunk 2]
```diff
@@ -33,22 +33,35 @@ export const useFormConductor = (stepName: string) => {
...
-      page,
-      limit,
-    }
-    return listingsService?.list(params)
+export interface FilterOptions {
+  preferredUnit?: string
+  accessibility?: string
+  community?: string
+  name?: string
+}
```
[the remark this answers is not in the document]
Comment (abbiefarr) [reply sites/public/lib/hooks.ts]: Added a TODO

**Review by avaleske — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**